### ISSUE TITLE
Windows support/python 3.7 issues? 

### ISSUE THREAD

**Issue by sgarriso** [tcprescott/pyz3r#1]
when running the code on windows using python 3.7 
I get an error with write rom. 
 def write_rom(self, rom, dstfilepath):
        fw = open(dstfilepath,"wb")
        patchrom = bytes()
        for idx, chunk_array in enumerate(self._chunk(rom,256)):
            patchrom += bytes(chunk_array)
        fw.write(patchrom)
        fw.close

Traceback (most recent call last):
  File "zelda.py", line 35, in <module>
    seed.write_rom(patched_rom, "patched_rom.sfc")
  File "C:\Users\linkm\AppData\Local\Programs\Python\Python37\lib\site-packages\pyz3r\__init__.py", line 324, in write_rom
    for idx, chunk_array in enumerate(self._chunk(rom,256)):
RuntimeError: generator raised StopIteration

I just wanted to check to see if this supports windows or if it is a 3.7 issue. Thanks!

**Comment by tcprescott** [comment]
I haven't tested it with Python 3.7, so it may have compatibility issues.
It should certainly work on Windows though.

I'll do some testing when I have some time.  Thanks for the report!

**Comment by sgarriso** [comment]
Thanks! it works on Bash on Windows with python 3.6. I want to believe that enumerate changed from 3.6 to 3.7  but I do not keep up with the change logs. I will be playing around with it later today so if I come up with a fix I will create a new branch and push it for ya. :) Also thank you for making this! I was going to try and do something like this but had no idea how to start! haha

**Comment by tcprescott** [comment]
Thanks!  This was the first python module that I published to pypi, so there is certainly room for improvement.

**Comment by tcprescott** [comment]
https://github.com/tcprescott/pyz3r/blob/1bca3bbb6cc2202594da670614cb8b6c1a69ef8a/pyz3r/__init__.py#L350

This should be changed from:

`yield tuple([next(itr) for i in range(count)])`

to

```
try:
    yield tuple([next(itr) for i in range(count)])
except StopIteration:
    return
```

I should be able to make this change and maintain compatibility with Python 3.6.

**Comment by sgarriso** [comment]
Okay Cool! I just tested that and that works! :)

**Comment by tcprescott** [comment]
I uploaded version 1.0.1 to PyPi, it has this fix for Python 3.7, plus it'll also retry if we don't get a response from the server (workaround for https://github.com/sporchia/alttp_vt_randomizer/issues/640 ).  It'll also use the /hash/ endpoint to retrieve seed data if it fails to pull from S3, allowing me to remove an ugly sleep(3) call.

Thanks for the bug report, it is much appreciated!
